### PR TITLE
Feature: TUI Config

### LINKED ISSUE #68: Add config for TUI
> I would also like a config for the cli itself so I can say like “default no borders” 

* look for a yaml file like `~/.astria/astria-go-conf.yaml` by default and also accept a file path to use a custom config.
* add `astria-go config get` and `astria-go config set` to get and set config options? is this overkill right now? it's also a pretty easy add
  * too easily confused with `astria-go dev config set` if we decide to create commands for managing the envars for the astria binaries?

┆Issue Number: ENG-162

### PR DESCRIPTION
**Pull request by sambukowski**
Adds a config file for the TUI app within the cli. This config enables setting values for the following app settings:
- Start with the AutoScroll setting enabled
- Start with word wrap enabled
- Have the fullscreen process panes start in borderless mode
- Set the default instance name
- Choose if the know services (cometbft, sequencer, composer, conductor) start minimized or not
- Choose if all generic services start minimized or not
- Allow override of the default highlight and border color of the process panes for color accessibility

closes #68

### CODE REVIEW

File: modules/cli/cmd/devrunner/config/tui.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+	output += fmt.Sprintf("GenericStartsMinimized: %v", c.GenericStartsMinimized)
+	output += fmt.Sprintf("HighlightColor: %s, ", c.HighlightColor)
+	output += fmt.Sprintf("BorderColor: %s", c.BorderColor)
+	output += "}"
+	return output
+}
+
+// LoadTUIConfigsOrPanic loads the TUIConfigs from the given path. If the file
+// cannot be loaded or parsed, the function will panic.
+func LoadTUIConfigsOrPanic(path string) TUIConfig {
```
Comment (steezeburger) [inline modules/cli/cmd/devrunner/config/tui.go:74]:
```suggestion
func LoadTUIConfigOrPanic(path string) TUIConfig {
```

File: modules/cli/cmd/devrunner/config/tui.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+	var config TUIConfig
+	if err := viper.Unmarshal(&config); err != nil {
+		log.Fatalf("Unable to decode into struct, %v", err)
+		panic(err)
+	}
+
+	return config
+}
+
+// CreateNetworksConfig creates a TUI configuration file and populates it
```
Comment (steezeburger) [inline modules/cli/cmd/devrunner/config/tui.go:91]:
```suggestion
// CreateTUIConfig creates a TUI configuration file and populates it
```

File: modules/cli/internal/ui/app.go [hunk 3]
```diff
@@ -45,9 +45,11 @@ func NewApp(processrunners []processrunner.ProcessRunner) *App {
 }
 
 // Start starts the tview application.
-func (a *App) Start() {
-	// create the state store for views to share ui state
-	stateStore := NewStateStore()
+func (a *App) Start(stateStore *StateStore) {
+	// if a state store wans't passed in, create a default one
```
Comment (steezeburger) [inline modules/cli/internal/ui/app.go:49]:
```suggestion
	// if a state store wasn't passed in, create a default one
```

File: modules/cli/internal/ui/statestore.go [hunk 4]
```diff
@@ -26,6 +26,17 @@ func NewStateStore() *StateStore {
 	}
 }
 
+// BuildStateStore creates a new StateStore with the given parameters
+func BuildStateStore(isAutoScroll, isWordWrap, isBorderless bool) *StateStore {
```
Comment (steezeburger) [inline modules/cli/internal/ui/statestore.go:30]: I think this should be named `NewStateStore` and the one without arguments `DefaultStateStore`, like you have `DefaultTUIConfig`

File: modules/cli/README.md [hunk 5]
```diff
@@ -189,11 +189,25 @@ configs and binaries for running the Astria stack.
 
 ## Configuration
 
-The CLI uses three configuration files:
+The CLI uses four configuration files:
```
Comment (steezeburger) [inline modules/cli/README.md:192]:
```suggestion
The CLI uses the following configuration files:
```

File: modules/cli/README.md [hunk 6]
```diff
@@ -189,11 +189,25 @@ configs and binaries for running the Astria stack.
...
+2. `base-config.toml`: Sets service environment variables
+3. `networks-config.toml`: Configures services and sequencer networks
+4. `sequencer-networks-config.toml`: Used for `astria-go sequencer` commands
+
+### Set TUI App Start State
+
+Once `astria-go dev init` has been run, edit `~/.astria/tui-config.toml` to
+control the starting settings of the TUI app.
+
+The `highlight_color` and `border_color` accept both named colors and
```
Comment (steezeburger) [inline modules/cli/README.md:204]: Is there a list of accepted color names anywhere?

**Review by steezeburger — CHANGES_REQUESTED** [review]

File: modules/cli/README.md (hunk 6 above)
Comment (sambukowski) [reply modules/cli/README.md:204]: Yes. It uses W3C colors: https://www.w3schools.com/tags/ref_colornames.asp

**Review by steezeburger — APPROVED** [review]

**Pull request merged**